### PR TITLE
Forms Automated Test - LPS-131360 | master

### PR DESCRIPTION
**Pull request by lucasperj**

### CODE REVIEW

**Comment by pedro-michel** [comment]
ci:test:sf

**Comment by pedro-michel** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase [hunk 1]
```diff
@@ -72,18 +70,55 @@ definition {
...
+			fieldPositionNumber = "0",
+			fieldType = "Select from List");
+
+		FormFields.editOptionsRow(
+			optionName = "Option1",
+			optionValue = "Option 1");
+
+		Check.checkToggleSwitch(
+			key_toggleSwitchLabel = "Required Field",
+			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
```
Comment (carolmariaabb) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase:102]: @lucasperj Can you use the existing macro `FormFields.enableSwitcher(fieldName = "Required Field");`?
Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase [hunk 2]
```diff
@@ -72,18 +70,55 @@ definition {
...
+
+		Form.save();
+
+		Form.publishForm();
+
+		FormsAdminNavigator.gotoPublishedForm();
+
+		Button.clickSubmitButton();
+
+		FormPortlet.submitWithValidationError(validationErrorMessage = "This field is required.");
```
Comment (carolmariaabb) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase:112]: @lucasperj the `submitWithValidationError` already clicks the submit button using the `FormPortlet.submit();` macro. I believe you don't need the `Button.clickSubmitButton();`. Can you please check? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase [hunk 3]
```diff
@@ -72,18 +70,55 @@ definition {
...
+
+		FormFields.selectSelectOption(
+			fieldName = "Select from List",
+			selectOption = "Option 1");
+
+		WaitForElementNotPresent(
+			locator1 = "Message#ERROR_FORM_FIELD_REQUIRED",
+			validationErrorMessage = "This field is required.");
```
Comment (carolmariaabb) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase:121]: @lucasperj Can you please use the existing macro `FormPortlet.waitForValidationErrorNotPresent(validationErrorMessage = "This field is required.");` instead? Thanks!

**Comment by carolmariaabb** [comment]
Hi @lucasperj , I didn’t find the `FormsSelectFromListField#CanBeRequired` test. Can you please check? Thanks!

**Comment by lucasperj** [comment]
> Hi @lucasperj , I didn’t find the `FormsSelectFromListField#CanBeRequired` test. Can you please check? Thanks!

Done!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase (hunk 3 above)
Comment (carolmariaabb) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase]: @lucasperj can you please check this comment?

Comment (lucasperj) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase]: Check!

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**